trpcultivate_phenotypes/src/Service/TripalCultivatePhenotypesTraitsService.php

```diff
@@ -255,163 +308,286 @@ public function insertTrait($trait, $schema = NULL) {
         'unit'  => $arr_trait['unit']['id']
       ];
     }
-    else {
-      return 0;
-    }
   }
 
   /**
    * Get trait.
    *
-   * @param array $trait
-   *   Key:
-   *     id - get trait by id number (cvterm_id) or
-   *     name - get trait by name (cvterm.name).
+   * @param string|int $trait
+   *   A string value is the trait name (cvterm.name), whereas an integer value
+   *   is the trait id number (cvterm.cvterm_id).
    *
    * @return object
-   *   Matching record/line in cvterm table.
+   *   Matching record/line in cvterm table or 0 if trait record Was not found.
+   *
+   * @dependencies
+   *   getTraitAsset()
    */
   public function getTrait($trait) {
-    // Configuration settings of the genus.
-    $genus_config = $this->config;
+    $trait_rec = $this->getTraitAsset($trait, 'trait');
+    return $trait_rec;
+  }
 
-    if (!isset($genus_config['trait']['id'])) {
-      // Genus is not configured.
+  /**
+   * Get trait method.
+   *
+   * @param string|int $trait
+   *   A string value is the trait name (cvterm.name), whereas an integer value
+   *   is the trait id number (cvterm.cvterm_id).
+   *
+   * @return array
+   *   All matching records (object) in an array. 0 if no methods were found.
```

Comment:
I don't like that these methods return 0 but rather think it should just return an empty array. This also makes it easier to check when using these functions IMO. For example, returning an empty array from these getters shrinks your code in the insert when checking if the method/unit already exists substantially. For example,

```diff
- if ($asset_rec) {
-   if (is_array($asset_rec)) {
-     foreach($asset_rec as $rec) {
-       if ($rec->cvterm_id == $object) {
-         $exists = TRUE;
-         break;
-       }
-    }
- }
+ $exists = FALSE;
+ foreach ($asset_rec as $rec) {
+   if ($rec->cvterm == $object) {
+     $exists = TRUE;
+     break;
+
+ }
```

Reply:
To be clear, I think the getTraitMethods() and getMethodUnits() should return empty array instead of 0. That is because they return sets of items.